validation/data-availability-type_test.go

```diff
@@ -0,0 +1,45 @@
+package validation
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/op-service/eth"
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func testDataAvailabilityType(t *testing.T, chain *ChainConfig) {
+	rpcEndpoint := Superchains[chain.Superchain].Config.L1.PublicRPC
+	require.NotEmpty(t, rpcEndpoint)
+
+	client, err := ethclient.Dial(rpcEndpoint)
+	require.NoErrorf(t, err, "could not dial rpc endpoint %s", rpcEndpoint)
+
+	// Standard chains must be eth-da for now. Will add checks for alt-da later
+	require.Equal(t, superchain.EthDA, chain.DataAvailabilityType)
+
+	batchSubmitterAddress := Addresses[chain.ChainID].BatchSubmitter
+	require.NotZero(t, batchSubmitterAddress)
+
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	depth := chain.SequencerWindowSize
+	require.NotZero(t, depth)
+
+	// First attempt without needing an archive node
+	//  - full nodes keep the last 128 blocks but we use 120 to give a buffer
+	nonArchivalDepth := 120
+	_, found, err := eth.CheckRecentTxs(ctx, client, nonArchivalDepth, common.Address(batchSubmitterAddress))
+	if !found {
+		_, found, err = eth.CheckRecentTxs(ctx, client, int(depth), common.Address(batchSubmitterAddress))
+	}
```

Comment:
what's the purpose of first only checking the non-archival range?

Reply:
Because locally I (and likely many other chain operators) do not have access to an archival node. However, our CI does

Reply:
We can leave it in since it doesn't hurt. But we could also just require devs and CIs that run this code to have access to an archival node, as a failing test on a full node is now inconclusive. I see that this actually works well for high-throughput chains, where batcher txs are frequent and probably below the 120 L1 blocks limit. But for low-throughput chains, which may be a probable case for new SCR arrivals, it could be misleading.

Note that you're currently not inspecting the first `err` assignment, since if `!found` you retry again. You might want to at least log the first `err`, if entering the `if !found` branch and `err` is non-nil.